ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/BaseLoadBalancer.java

```diff
@@ -603,6 +603,52 @@ public void cancelPingTask() {
     }
 
     /**
+     * Template method, which allows <c>BaseLoadBalancer</c> to use custom
+     * <c>IPingerStrategy</c>. Default implementation performs ping serially,
+     * which may not be applicable, if your <c>IPing</c> implementation is
+     * slow, or you have large number of servers.
+     */
+    protected IPingerStrategy newPingerStrategy() {

```

Comment:
I'd rather see a similar configuration to `IPing` and `IRule`. I don't think it's practical to extend `BaseLoadBalancer` in many cases. It would be easier to have it in a constructor and only call this method if there isn't a custom on passed in.